File: logger/src/test/java/com.orhanobut.logger/DiskLogStrategyTest.java
package com.orhanobut.logger;

import android.os.Handler;

import org.junit.Test;

import static com.orhanobut.logger.Logger.DEBUG;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

public class DiskLogStrategyTest {

  @Test public void log() {
    Handler handler = mock(Handler.class);
    LogStrategy logStrategy = new DiskLogStrategy(handler);

    logStrategy.log(DEBUG, "tag", "message");

    verify(handler).sendMessage(handler.obtainMessage(DEBUG, "message"));
  }

}

File: logger/src/test/java/com.orhanobut.logger/DiskLogStrategyTest.kt
package com.orhanobut.logger

import android.os.Handler

import org.junit.Test

import com.orhanobut.logger.Logger.DEBUG
import org.mockito.Mockito.mock
import org.mockito.Mockito.verify

class DiskLogStrategyTest {

  @Test fun log() {
    val handler = mock(Handler::class.java)
    val logStrategy = DiskLogStrategy(handler)

    logStrategy.log(DEBUG, "tag", "message")

    verify(handler).sendMessage(handler.obtainMessage(DEBUG, "message"))
  }

}